test/tests/misc/solution_invalid/tests

```diff
@@ -9,15 +9,34 @@
   []
 
   [allow_solution_invalid]
-    type = 'RunException'
+    type = 'RunApp'
     input = 'solution_invalid.i'
     cli_args = 'Problem/allow_invalid_solution=true'
-    expect_err = 'The Solution Invalidity warnings are detected but silenced! Use Problem/allow_invalid_solution=false to activate'
     design = 'SolutionInvalidity.md'
     issues = '#22814'
     requirement = 'The system shall allow overriding the solution validity checks.'
   []
 
+  [solution_invalid_warning]
+    design = 'SolutionInvalidity.md'
+    issues = '#22814'
+    requirement = 'The system shall be able to count warnings through the solution invalidity interface and report them,'
+    [no_console_output]
+      type = 'RunException'
+      input = 'solution_invalid.i'
+      cli_args = 'Materials/filter/flag_solution_warning=true Problem/show_invalid_solution_console=false'
+      expect_err = 'The Solution Invalidity warnings are detected but silenced! Use Problem/show_invalid_solution_console=true to show solution counts'
+      detail = 'unless the user specifies to silence them,'
+    []
+    [console_output]
+      type = 'JSONDiff'
+      input = 'solution_invalid.i'
+      cli_args = "Materials/filter/flag_solution_warning=true Outputs/file_base='solution_invalid_warning'"
+      jsondiff = 'solution_invalid_warning.json'
+      detail = 'and otherwise both in the console window and a json file.'
```

Comment:
you could add an expect_out to test that it appears in the console (in the same test as JSONDiff)